x-pack/plugins/ml/public/application/explorer/add_to_dashboard_control.tsx

```diff
@@ -0,0 +1,334 @@
+        setAddedDashboards({ ...addedDashboards, [item.id]: 'success' });
+      } catch (e) {
+        toasts.danger({
+          body: e,
+        });
+      }
+    },
+    [addedDashboards, selectedSwimlanes]
+  );
+
+  const columns: EuiTableProps['columns'] = useMemo(
+    () => [
+      {
+        field: 'title',
+        name: i18n.translate('xpack.ml.explorer.dashboardsTable.titleColumnHeader', {
+          defaultMessage: 'Title',
+        }),
+        sortable: true,
+        truncateText: true,
+      },
+      {
+        field: 'description',
+        name: i18n.translate('xpack.ml.explorer.dashboardsTable.descriptionColumnHeader', {
+          defaultMessage: 'Description',
+        }),
+        truncateText: true,
+      },
+      {
+        field: 'actions',
+        name: i18n.translate('xpack.ml.explorer.dashboardsTable.actionsColumnHeader', {
+          defaultMessage: 'Actions',
+        }),
+        align: 'right',
+        actions: [
+          {
+            name: i18n.translate('xpack.ml.explorer.dashboardsTable.quickAddLabel', {
+              defaultMessage: 'Quick add',
+            }),
+            render: (item) => {
+              return addedDashboards[item.id] === 'pending' ? (
+                <EuiLoadingSpinner size="m" />
+              ) : (
+                <EuiToolTip
+                  content={
+                    addedDashboards[item.id] === 'success' ? (
+                      <FormattedMessage
+                        id="xpack.ml.explorer.dashboardsTitle.successfullyAddedMessage"
+                        defaultMessage="Successfully added"
+                      />
+                    ) : (
+                      <FormattedMessage
+                        id="xpack.ml.explorer.dashboardsTable.quickAddLabel"
+                        defaultMessage="Quick add"
+                      />
+                    )
+                  }
+                >
+                  <EuiButtonIcon
+                    color="primary"
+                    onClick={addSwimlaneToDashboardCallback.bind(null, item)}
+                    iconType={
+                      addedDashboards[item.id] === 'success'
+                        ? 'checkInCircleFilled'
+                        : 'plusInCircle'
+                    }
+                    aria-label="Quick add"
+                    disabled={addedDashboards[item.id] === 'success'}
+                  />
+                </EuiToolTip>
+              );
+            },
+          },
+          {
+            name: i18n.translate('xpack.ml.explorer.dashboardsTable.editInDashboard', {
+              defaultMessage: 'Edit in dashboard',
+            }),
+            onClick: async (item) => {
+              await addSwimlaneToDashboardCallback(item);
+              setAddedDashboards({ ...addedDashboards, [item.id]: 'redirecting' });
+              window.location.href = await dashboardService.getDashboardEditUrl(item.id);
```

Comment:
I believe this causes a full page reload? 
You could try using platforms `navigateToApp()` api in application service to navigate without full page reload. 
Similar example: https://github.com/elastic/kibana/blob/master/x-pack/plugins/dashboard_enhanced/public/services/drilldowns/dashboard_to_dashboard_drilldown/drilldown.tsx#L72

Reply:
thanks, good point! wasn't aware of this function, fixed in https://github.com/elastic/kibana/pull/68784/commits/0a6cdc69fe3ce5730321ff22c78a5c1e8a29a849